Fix/#623 fixed email validation in users details

Patrykb0802 opened the pull request.

Commit:
#623 Fixed email validation in users details
- Added class: EmailValidator used in User.validate

Commit:
#623 Fixed email validation in users details
- Added test for MailValidator: EmailValidatorTest.java

Limraj commented on test/com/serotonin/mango/util/EmailValidatorTest.java:
```diff
@@ -0,0 +1,57 @@
+
+    @Test
+    public void when_isValidEmail() {
+        //given:
```
Can be removed

Limraj commented on test/com/serotonin/mango/util/EmailValidatorTest.java:
```diff
@@ -0,0 +1,57 @@
+@RunWith(Parameterized.class)
+public class EmailValidatorTest {
+
+    @Parameterized.Parameters
```
(name= "{index}: email: {0}, valid: {1}")

Commit:
#623 Fixed email validation in users details
- Validation fix, currently lets emails starting with a capital letter through
- Added detailed tests output

Commit:
#623 Fixed email validation in users details
- removed unnecesary comments
